src/rauc-hawkbit-updater.c

```diff
@@ -155,8 +159,12 @@ int main(int argc, char **argv)
 
         struct config *config = load_config_file(config_file, &error);
         if (config == NULL) {
-                g_printerr("Loading config file failed: %s\n", error->message);
-                g_error_free(error);
+                if (error != NULL) {
```

Comment:
The same as noted above applies here too. With the difference that `load_config_file()` can actually return NULL without setting GError properly. But this should be fixed in `load_config_file()`, not with a hack :smirk:

Reply:
For sure, in general, it is not a bad idea to protect against NULL pointers.

But the check is wrong here as in case of a failure, the GError must contain a valid error and `NULL` is the indicator for 'no error'. Thus we use the condition actually indicating 'no error' to say 'there was an error we do not know about'. This is inconsistent and hides issues. [1]

This could be called a 'hack' or a 'bug', but anyway, it is no appropriate handling of GError. ;)

[1] https://developer.gnome.org/glib/stable/glib-Error-Reporting.html